index.js

```diff
@@ -0,0 +1,77 @@
+'use babel';
+
+import path from 'path';
+import { execNode, find } from 'atom-linter';
+
+const GRAMMAR_SCOPES = [
+  'text.html.angular',
+  'text.html.basic',
+  'text.html.erb',
+  'text.html.gohtml',
+  'text.html.jsp',
+  'text.html.mustache',
+  'text.html.handlebars',
+  'text.html.ruby'
+];
+
+export const config = {
+  executablePath: {
+    title: 'Executable Path',
+    description: 'HTMLHint Executable Path',
+    type: 'string',
+    default: path.join(__dirname, '..', 'node_modules', 'htmlhint', 'bin', 'htmlhint')
+  }
+};
+
+const executablePath = () => atom.config.get('linter-htmlhint.executablePath');
+
+export const activate = () => {
+  require('atom-package-deps').install('linter-htmlhint');
+};
+
+export const provideLinter = () => {
+  return {
+    name: 'htmlhint',
+    grammarScopes: GRAMMAR_SCOPES,
+    scope: 'file',
+    lintOnFly: true,
+    lint: editor => {
+      const text = editor.getText();
+      const filePath = editor.getPath();
+
+      if (!text) {
+        return [];
+      }
+
+      const parameters = [filePath, '--format', 'json'];
+      const htmlhintrc = find(filePath, '.htmlhintrc');
+
+      if (htmlhintrc) {
+        parameters.push('-c');
+        parameters.push(htmlhintrc);
+      }
+
+      return execNode(executablePath(), parameters, {}).then(output => {
+        const results = JSON.parse(output);
+
+        if (!results.length) {
+          return [];
+        }
+
+        const messages = results[0].messages;
+
+        return messages.map(message => {
+          return {
+            range: [
```

Comment:
Use `helpers.rangeFromLineNumber()` if `htmlhint` doesn't return an endpoint for its messages. This way instead of your messages being a single character the rest of the line is highlighted meaning it's easier for the user to see there is an issue.